app/src/main/java/com/example/weather/MainActivity.java

```diff
@@ -2,13 +2,192 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.net.Uri;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import javax.net.ssl.HttpsURLConnection;
 
 public class MainActivity extends AppCompatActivity {
 
+    private TextView widgetViewWeather;
+    private EditText widgetEditCity;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        widgetViewWeather = findViewById(R.id.msg_temperature);
+        widgetEditCity = findViewById(R.id.msg_city);
+    }
+
+    public void onClickByCity(View view) throws MalformedURLException {
```

Comment:
this method shouldn't throw exception because app can crash. Handle exception inside.